libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1153,16 +1153,6 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         }
     }
 
-    private fun resetTripSessionRoutes() {
-        threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
-            routeUpdateMutex.withLock {
-                val routes = directionsSession.routes
-                val legIndex = latestLegIndex ?: directionsSession.initialLegIndex
-                setRoutesToTripSession(routes, SetRoutes.NewRoutes(legIndex))
-            }
-        }
-    }
```

Comment:
why was it needed? 🤔

Reply:
I believe there were cases when the routes might have been present on the NavSDK side, but not on NN side. Or it was added just in case, because we didn't know exactly how NN works with routes.